src/Microsoft.IdentityModel.Tokens/RsaSecurityKey.cs

```diff
@@ -165,5 +190,39 @@ public override int KeySize
         /// <see cref="RSA"/> instance used to initialize the key.
         /// </summary>
         public RSA Rsa { get; private set; }
+
+        private RSAParameters RemoveLeadingZero(RSAParameters rsaParameters)
+        {
+            // Sometimes the parameters we received have a leading 0, the reason is the parameters 
+            // are positive integers, the generator of RSA key may put a leading 0 as the sign
+            // digit. RSACng will ignore the leading 0, but RSACryptoServiceProvider doesn't, so
+            // we have to remove the leading 0 for .NET framework.
+            rsaParameters.Modulus = RemoveLeadingZeroInBase64UrlDecodedBytes(rsaParameters.Modulus);
+            rsaParameters.Exponent = RemoveLeadingZeroInBase64UrlDecodedBytes(rsaParameters.Exponent);
+            rsaParameters.D = RemoveLeadingZeroInBase64UrlDecodedBytes(rsaParameters.D);
+            rsaParameters.P = RemoveLeadingZeroInBase64UrlDecodedBytes(rsaParameters.P);
+            rsaParameters.Q = RemoveLeadingZeroInBase64UrlDecodedBytes(rsaParameters.Q);
+            rsaParameters.DP = RemoveLeadingZeroInBase64UrlDecodedBytes(rsaParameters.DP);
+            rsaParameters.DQ = RemoveLeadingZeroInBase64UrlDecodedBytes(rsaParameters.DQ);
+            rsaParameters.InverseQ = RemoveLeadingZeroInBase64UrlDecodedBytes(rsaParameters.InverseQ);
+            return rsaParameters;
+        }
+
+        private byte[] RemoveLeadingZeroInBase64UrlDecodedBytes(byte[] bytes)
+        {
+            if (bytes == null)
+                return bytes;
+
+            var n = bytes.Length;
+            if (n > 0 && bytes[0].Equals(0x00))
```

Comment:
What happens if n==1?

Reply:
all rsa parameters are positive integers, if n==1, the byte cannot be 0, so this is fine. I changed the condition to n>1 in the add fix anyway.